src/ChronosTime.php

```diff
@@ -0,0 +1,121 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @license       https://www.opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\Chronos;
+
+use DateTimeInterface;
+use InvalidArgumentException;
+
+class ChronosTime
+{
+    /**
+     * @var int
+     */
+    protected const MICRO_PER_SECOND = 1000000;
+
+    /**
+     * @var int
+     */
+    protected const MICRO_PER_MINUTE = self::MICRO_PER_SECOND * 60;
+
+    /**
+     * @var int
+     */
+    protected const MICRO_PER_HOUR = self::MICRO_PER_MINUTE * 60;
+
+    /**
+     * @var int
+     */
+    protected const MICRO_PER_DAY = self::MICRO_PER_HOUR * 24;
+
+    /**
+     * @var int
+     */
+    protected int $microseconds;
+
+    /**
+     * @param \Cake\Chronos\Chronos|\Cake\Chronos\ChronosDate|\Cake\Chronos\ChronosTime|\DateTimeInterface|string|int $time Time string or microseconds
+     */
+    public function __construct(Chronos|ChronosDate|ChronosTime|DateTimeInterface|string|int $time = 0)
```

Comment:
```suggestion
    public function __construct(Chronos|ChronosTime|DateTimeInterface|string|int $time = 0)
```

I don't know how you could convert a Date into Time.